src/main/java/net/dv8tion/jda/internal/interactions/component/TextInputImpl.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.interactions.component;
+
+import net.dv8tion.jda.api.interactions.components.text.TextInput;
+import net.dv8tion.jda.api.interactions.components.text.TextInputStyle;
+import net.dv8tion.jda.api.utils.data.DataObject;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class TextInputImpl implements TextInput
+{
+    private final String id;
+    private final TextInputStyle style;
+    private final String label;
+    private final int minLength;
+    private final int maxLength;
+    private final boolean required;
+    private final String value;
+    private final String placeholder;
+
+    public TextInputImpl(DataObject object)
+    {
+        this(
+                object.getString("custom_id"),
+                TextInputStyle.fromKey(object.getInt("style", -1)),
+                object.getString("label", null),
+                object.getInt("min_Length", -1),
+                object.getInt("max_length", -1),
+                object.getBoolean("required", false),
+                object.getString("value", null),
+                object.getString("placeholder", null)
+        );
+    }
+
+    public TextInputImpl(
+            String id, TextInputStyle style, String label, int minLength,
+            int maxLength, boolean required, String value, String placeholder)
+    {
+        this.id = id;
+        this.style = style;
+        this.label = label;
+        this.minLength = minLength;
+        this.maxLength = maxLength;
+        this.required = required;
+        this.value = value;
+        this.placeholder = placeholder;
+    }
+
+    @Nonnull
+    @Override
+    public Type getType()
+    {
+        return Type.TEXT_INPUT;
+    }
+
+    @Nonnull
+    @Override
+    public TextInputStyle getStyle()
+    {
+        return style;
+    }
+
+    @Nonnull
+    @Override
+    public String getId()
+    {
+        return id;
+    }
+
+    @Nonnull
+    @Override
+    public String getLabel()
+    {
+        return label;
+    }
+
+    @Override
+    public int getMinLength()
+    {
+        return minLength;
+    }
+
+    @Override
+    public int getMaxLength()
+    {
+        return maxLength;
+    }
+
+    @Override
+    public boolean isRequired()
+    {
+        return required;
+    }
+
+    @Nullable
+    @Override
+    public String getValue()
+    {
+        return value;
+    }
+
+    @Nullable
+    @Override
+    public String getPlaceHolder()
+    {
+        return placeholder;
+    }
+
+    @Nonnull
+    @Override
+    public DataObject toData()
+    {
+        DataObject obj = DataObject.empty()
+                    .put("type", 4)
```

Comment:
We should expose `Component.Type#getKey()` for the following use
```suggestion
                    .put("type", Type.TEXT_INPUT.getKey())
```
or
```suggestion
                    .put("type", getType().getKey())
```